### PR TITLE
CU-86873z3fk Add models

### PR DESCRIPTION
**Pull request by ImMPrada**
This PR adds migrations, models an spec for this first phase of the DB:

image

### CODE REVIEW

File: app/models/registered_url.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class RegisteredUrl < ApplicationRecord
+  has_many :visits, dependent: :destroy
```
Comment (mapra99) [inline app/models/registered_url.rb:2]: are you going to use a counter cache to count the number of visits of each URL? I think we had talked something about that, right? https://blog.appsignal.com/2018/06/19/activerecords-counter-cache.html

**Review by mapra99 — COMMENTED** [review]
This looks good! I left some comments

File: app/models/registered_url.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def self.create_registered_url(temporary_session, url)
+    registered_url = new(url:, uuid: SecureRandom.hex(6))
+    registered_url.temporary_session = temporary_session
+
+    return registered_url if registered_url.save
+
+    create_registered_url(temporary_session, url)
+  end
```
[the remark this answers is not in the document]
Comment (ImMPrada) [reply app/models/registered_url.rb]: done based on our discussion!

File: app/models/temporary_session.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  has_many :registered_urls
+
+  validates :uuid, presence: true, uniqueness: true
+
+  def self.create_temporary_session
+    temporary_session = new(uuid: UUID7.generate)
+    return temporary_session if temporary_session.save
+
+    create_temporary_session
+  end
```
[the remark this answers is not in the document]
Comment (ImMPrada) [reply app/models/temporary_session.rb]: done based on our discussion!

File: app/models/registered_url.rb (hunk 1 above)
Comment (ImMPrada) [reply app/models/registered_url.rb]: yeah, done!

I needed to add a column "visits_count" to registered_urls, and enable cache  in Visit model

**Pull request merged**